lib/msal-core/src/authority/AuthorityFactory.ts

```diff
@@ -25,6 +29,32 @@ export class AuthorityFactory {
         }
     }
 
+    public static async resolveAuthorityAsync(authorityInstance: Authority, telemetryManager: TelemetryManager, correlationId: string): Promise<ITenantDiscoveryResponse> {
```

Comment:
Can we brainstorm other names for this and `parseAuthorityMetadata` to more clearly describe what they are doing and how they are related? Something along the lines of `saveMetadataFromNetwork` and `saveMetadataFromConfig`?

Reply:
This is a little too similar to `resolveEndpointsAsync` and it took me a minute to understand how these differ and why this needs to be in AuthorityFactory

Reply:
+1 for naming.

Reply:
Renamed based on feedback.